docs/02-app/01-building-your-application/07-configuring/01-typescript.mdx

```diff
@@ -70,6 +72,20 @@ The TypeScript plugin can help with:
 
 It is highly recommended to be on at least `v4.5.2` of TypeScript to get syntax features such as [type modifiers on import names](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names) and [performance improvements](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#real-path-sync-native).
 
+## Type checking with `next.config.ts`
+
+You can use TypeScript in your Next.js configuration by using `next.config.ts`. If you have a `next.config.js` or `next.config.mjs` file, you can rename it to `next.config.ts`, and will be able to use TypeScript.
+
+```ts filename="next.config.ts"
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  /* config options here */
+}
+
+export default nextConfig
+```
+
```

Comment:
Maybe move the second sentence into a **good to know** note. Sorry, the comment is messed up because of the code block. 

```suggestion
You can use TypeScript in your Next.js configuration by using `next.config.ts`.

```ts filename="next.config.ts"
import type { NextConfig } from 'next'

const nextConfig: NextConfig = {
  /* config options here */
}

export default nextConfig
```
> **Good to know**:  If you have a `next.config.js` or `next.config.mjs` file, you can safely rename it to `next.config.ts`.
```